### PR TITLE
Abdullah

### PR DESCRIPTION
**Pull request by AbdullahMarikkar**
[ITP-1785] 
[ITP-1790]
[ITP-1795]
[ITP-1800]
[ITP-1805]
[ITP-1810]

[ITP-1759]: https://fcodelabs.atlassian.net/browse/ITP-1759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1760]: https://fcodelabs.atlassian.net/browse/ITP-1760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1761]: https://fcodelabs.atlassian.net/browse/ITP-1761?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ITP-1762]: https://fcodelabs.atlassian.net/browse/ITP-1762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ITP-1763]: https://fcodelabs.atlassian.net/browse/ITP-1763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1764]: https://fcodelabs.atlassian.net/browse/ITP-1764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ITP-1785]: https://fcodelabs.atlassian.net/browse/ITP-1785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1790]: https://fcodelabs.atlassian.net/browse/ITP-1790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1795]: https://fcodelabs.atlassian.net/browse/ITP-1795?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1800]: https://fcodelabs.atlassian.net/browse/ITP-1800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1805]: https://fcodelabs.atlassian.net/browse/ITP-1805?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-1810]: https://fcodelabs.atlassian.net/browse/ITP-1810?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by udaraJ08 — APPROVED** [review]

File: dear-diary-react/src/components/DiaryForm.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  const handleChange2 = (event: { target: { value: any } }) => {
+    setInputValue2(event.target.value);
+  };
+  const handleSubmit = (event: { preventDefault: () => void }) => {
+    event.preventDefault();
+    // Do something with the input value
+    inputValueOne = inputValue;
+    inputValueTwo = inputValue2;
+    cardInputs.push({ inputValueOne, inputValueTwo });
+    console.log(cardInputs);
```
Comment (udaraJ08) [inline dear-diary-react/src/components/DiaryForm.tsx]: Remove unnecessary console.logs()

File: dear-diary-react/src/components/DiaryForm.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import Typography from "@mui/material/Typography";
+import "../index.css";
+
+let inputValueOne = "";
+let inputValueTwo = "";
+const cardInputs: any[] = [];
```
Comment (udaraJ08) [inline dear-diary-react/src/components/DiaryForm.tsx]: Do not use any. Define a custom data type using an interface

**Review by udaraJ08 — CHANGES_REQUESTED** [review]

File: dear-diary-react/src/Containers/SignInFiles/SignIn.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import logo from "../../logo.svg";
+import "../../App.css";
+import RandomButton from "../../components/Buttons/RandomButton";
+import Box from "@mui/material/Box";
+import { Grid, Typography } from "@mui/material";
+
+function SignIn() {
+  return (
+    <div
```
Comment (shamal-iroshan) [inline dear-diary-react/src/Containers/SignInFiles/SignIn.tsx]: don't need to write components like this, use mui components and there way of styling

File: dear-diary-react/src/Containers/SignInFiles/SignIn.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                fontWeight: "lighter",
+                marginTop: "6px",
+              }}
+            >
+              Dear Diary
+            </Typography>
+          </Grid>
+        </div>
+        <div className="App">
+          <h1
```
Comment (shamal-iroshan) [inline dear-diary-react/src/Containers/SignInFiles/SignIn.tsx:60]: here as well, change this also to a MUI component

File: dear-diary-react/src/components/AppRouter.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+import SignIn from "../Containers/SignInFiles/SignIn";
+import DiaryHome from "../Containers/Diary/DiaryHome";
+
+function AppRouter() {
+  return (
+    <Router>
+      <Routes>
+        <Route path="/" element={<SignIn />} />
+        <Route path="/DiaryHome" element={<DiaryHome />} />
```
Comment (shamal-iroshan) [inline dear-diary-react/src/components/AppRouter.tsx]: you can use constants to store these paths and use it when ever you want

File: dear-diary-react/src/components/Buttons/RandomButton.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  "Ava",
+  "Lucas",
+  "Mia",
+  "Benjamin",
+  "Evelyn",
+  "Theodore",
+  "Luna",
+];
+
+export const generateRandomName = function () {
```
Comment (shamal-iroshan) [inline dear-diary-react/src/components/Buttons/RandomButton.tsx]: move this random name generation function to a utility file

File: dear-diary-react/src/components/DiaryComponents/DiaryForm.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+import CardContent from "@mui/material/CardContent";
+import Typography from "@mui/material/Typography";
+import "../../index.css";
+import { useDispatch } from "react-redux";
+import { addCard } from "../../app/cardSlice";
+import { Cards } from "../../app/cardSlice";
+import { useSelector } from "react-redux";
+import { RootState } from "../../app/store";
+
+// let inputValueOne = "";
```
Comment (shamal-iroshan) [inline dear-diary-react/src/components/DiaryComponents/DiaryForm.tsx]: if not needed remove these commented lines

File: dear-diary-react/src/components/DiaryComponents/DiaryForm.tsx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+            value={inputValue2}
+            label="Enter Description"
+            multiline
+            rows={4}
+            sx={{ width: "100%", marginTop: "10px" }}
+          />
+        </form>
+      </Box>
+      <ul className="pizzas">
+        {/* {showCard &&
```
Comment (shamal-iroshan) [inline dear-diary-react/src/components/DiaryComponents/DiaryForm.tsx]: if not needed remove these commented lines

File: dear-diary-react/src/components/DiaryComponents/DiaryForm.tsx [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+              description={card.description}
+              key={card.title}
+            />
+          ))}
+      </ul>
+    </div>
+  );
+};
+
+export function RecipeReviewCard(props: Cards) {
```
Comment (shamal-iroshan) [inline dear-diary-react/src/components/DiaryComponents/DiaryForm.tsx:129]: if this is a component move it to the components section

**Review by shamal-iroshan — CHANGES_REQUESTED** [review]
fix commented things

**Pull request merged**